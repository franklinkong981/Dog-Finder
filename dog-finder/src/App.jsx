import React from 'react';
import {Routes, Navigate, Route} from "react-router-dom";
import './App.css';

import DogList from "./DogList.jsx";
import FilterDogDetails from "./FilterDogDetails.jsx";

import Whiskey from "./assets/images/whiskey.jpg";
import Perry from "./assets/images/perry.jpg";
import Duke from "./assets/images/duke.jpg";
import Tubby from "./assets/images/tubby.jpg";

function App({dogsInfo}) {

  return (
   <div className="App">
    {/*<Nav/>*/}
    <Routes>
      <Route path="/dogs" element={<DogList dogsInfo={dogsInfo}/>} />
      <Route path="/dogs/:name" element={<FilterDogDetails allDogsInfo={dogsInfo}/>} />
      <Route path="*" element={<Navigate to="/dogs" />} />
    </Routes>
   </div>
  );
}

App.defaultProps = {
  dogsInfo: [
    {
      name: "Whiskey",
      age: 5,
      src: Whiskey,
      facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
      ]
    },
    {
      name: "Duke",
      age: 3,
      src: Duke,
      facts: [
        "Duke believes that ball is life.",
        "Duke likes snow",
        "Duke enjoys pawing other dogs."
      ]
    },
    {
      name: "Perry",
      age: 4,
      src: Perry,
      facts: [
        "Perry loves all humans.",
        "Perry demolishes all snacks.",
        "Perry hates the rain."
      ]
    },
    {
      name: "Tubby",
      age: 4,
      src: Tubby,
      facts: [
        "Tubby is really stupid.",
        "Tubby does not like walks.",
        "Angelina used to hate Tubby, but claims to anymore."
      ]
    }
  ]
};

export default App;
