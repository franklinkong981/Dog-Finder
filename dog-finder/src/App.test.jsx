import React from "react";
import App from "./App";
import {render, fireEvent} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";
import { describe, expect, it } from "vitest";

describe("Test <App/> rendering", function () {
  it("renders without crashing", function() {
    render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));
  });
  it ("matches the snapshot for the MadLibs component", function() {
    const {asFragment} = render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Tests functionality of the App component with routing", function() {
  it("Makes sure the links on the page work", function() {
    const {getByText} = render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));
    
    expect(getByText("Welcome to the Dog Finder!")).toBeInTheDocument();
  
    //tests a link. Does clicking on Whiskey link take you to Whiskey details page, is there a button there that takes you back to the home page?
    let whiskeyLink = getByText("Whiskey - 5 years old");
    expect(whiskeyLink).toBeInTheDocument();
    fireEvent.click(whiskeyLink);
  
    //should be on Whiskey details page now.
    expect(whiskeyLink).not.toBeInTheDocument();
    const whiskeyFunFact = getByText("Whiskey loves eating popcorn.");
    expect(whiskeyFunFact).toBeInTheDocument();
    const backToHomeButton = getByText("Back to Home Page");
    expect(backToHomeButton).toBeInTheDocument();
    fireEvent.click(backToHomeButton);
  
    //should be back at the home page now.
    whiskeyLink = getByText("Whiskey - 5 years old");
    expect(whiskeyLink).toBeInTheDocument();
  });

  it("Makes sure the navigation links work", function() {
    const {getByText} = render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));

    const perryNavLink = getByText("Perry");
    expect(perryNavLink).toBeInTheDocument();
    const tubbyNavLink = getByText("Tubby");
    expect(tubbyNavLink).toBeInTheDocument();
    fireEvent.click(perryNavLink);

    //We should now be on Perry's page.
    expect(perryNavLink).toBeInTheDocument();
    expect(tubbyNavLink).toBeInTheDocument();
    const perryFunFact = getByText("Perry loves all humans.");
    expect(perryFunFact).toBeInTheDocument();
    fireEvent.click(tubbyNavLink);

    //We should now be on Tubby's page.
    expect(perryNavLink).toBeInTheDocument();
    expect(tubbyNavLink).toBeInTheDocument();
    const tubbyFunFact = getByText("Tubby is really stupid.");
    expect(tubbyFunFact).toBeInTheDocument();
  });
});

