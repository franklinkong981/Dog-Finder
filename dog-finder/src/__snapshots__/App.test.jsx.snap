// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test <App/> rendering > matches the snapshot for the MadLibs component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="Nav"
    >
      <a
        class="inactive"
        data-discover="true"
        href="/dogs/Whiskey"
      >
        Whiskey
      </a>
      <a
        class="inactive"
        data-discover="true"
        href="/dogs/Duke"
      >
        Duke
      </a>
      <a
        class="inactive"
        data-discover="true"
        href="/dogs/Perry"
      >
        Perry
      </a>
      <a
        class="inactive"
        data-discover="true"
        href="/dogs/Tubby"
      >
        Tubby
      </a>
    </nav>
    <div
      class="DogList"
    >
      <h1
        class="DogList-header"
      >
        Welcome to the Dog Finder!
      </h1>
      <h2
        class="DogList-header-2"
      >
        List of dogs:
      </h2>
      <div
        class="DogList-dog"
      >
        <img
          alt="Picture of Whiskey"
          class="DogList-dog-picture"
          src="/src/assets/images/whiskey.jpg"
        />
        <br />
        <a
          class="DogList-dog-name"
          data-discover="true"
          href="/dogs/Whiskey"
        >
          Whiskey - 5 years old
        </a>
        <hr />
      </div>
      <div
        class="DogList-dog"
      >
        <img
          alt="Picture of Duke"
          class="DogList-dog-picture"
          src="/src/assets/images/duke.jpg"
        />
        <br />
        <a
          class="DogList-dog-name"
          data-discover="true"
          href="/dogs/Duke"
        >
          Duke - 3 years old
        </a>
        <hr />
      </div>
      <div
        class="DogList-dog"
      >
        <img
          alt="Picture of Perry"
          class="DogList-dog-picture"
          src="/src/assets/images/perry.jpg"
        />
        <br />
        <a
          class="DogList-dog-name"
          data-discover="true"
          href="/dogs/Perry"
        >
          Perry - 4 years old
        </a>
        <hr />
      </div>
      <div
        class="DogList-dog"
      >
        <img
          alt="Picture of Tubby"
          class="DogList-dog-picture"
          src="/src/assets/images/tubby.jpg"
        />
        <br />
        <a
          class="DogList-dog-name"
          data-discover="true"
          href="/dogs/Tubby"
        >
          Tubby - 4 years old
        </a>
        <hr />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
